ts/client/src/utils.ts

```diff
@@ -218,3 +218,36 @@ export declare abstract class As<Tag extends keyof never> {
   private static readonly $as$: unique symbol;
   private [As.$as$]: Record<Tag, true>;
 }
+
+export function deepClone<T>(obj: T, hash = new WeakMap()): T {
```

Comment:
do you have a reference for this copy paste?

Reply:
is it from stackoverflow?

Reply:
Nope combined few methods from multiple sources when looking around for solution.

Reply:
why is it faster than lodash?

Reply:
Health Objects are mostly I80F48 or BN, lodash is well optimized (fast-clone lib is faster added it also in this pr for group object cloning) and have custom implementation to cover corner cases, it will run more operations on objects like BN to clone them because it treat it as more complex then it really is. So when you just want to get rid of reference to do some math on those its enough to run native js functions to copy them i also made performance checks and its indeed faster. Also lodash don't preserve prototype chain like this method has but in case of BN it won't affect math operations it would be problem when we would have more nested classes inside this object with functions that change values but for that case fast-copy is there.

https://www.npmjs.com/package/fast-copy have table for performance of different libs also runned my own with react tools.